javascript/index.js

```diff
@@ -23,18 +23,153 @@
+    getInstruction(
+      "mashedPotatoes",
+      1,
+      (step1) => {
+        document.querySelector(
+          "#mashedPotatoes"
+        ).innerHTML += `<li>${step1}</li>`;
+        getInstruction(
+          "mashedPotatoes",
+          2,
+          (step2) => {
+            document.querySelector(
+              "#mashedPotatoes"
+            ).innerHTML += `<li>${step2}</li>`;
+            getInstruction(
+              "mashedPotatoes",
+              3,
+              (step3) => {
+                document.querySelector(
+                  "#mashedPotatoes"
+                ).innerHTML += `<li>${step3}</li>`;
+                getInstruction("mashedPotatoes", 4, (step4) => {
+                  document.querySelector(
+                    "#mashedPotatoes"
+                  ).innerHTML += `<li>${step4}</li>`;
+
+                  const mpReady = document.createElement("li");
+                  mpReady.innerHTML = `Mashed potatoes are ready!`;
+                  document
+                    .querySelector("#mashedPotatoes")
+                    .appendChild(mpReady);
+
+                  document
+                    .querySelector("#mashedPotatoesImg")
+                    .removeAttribute("hidden");
+                });
+              },
+              (error) => console.log(error)
+            );
+          },
+          (error) => console.log(error)
+        );
+      },
+      (error) => console.log(error)
+    );
+  },
+  (error) => console.log(error)
+);
+
+
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0)
+  .then((step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+    //  ... Your code here
+    return obtainInstruction("steak", 1);
+  })
+  .then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    return obtainInstruction("steak", 2);
+  })
+  .then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    return obtainInstruction("steak", 3);
+  })
+  .then((step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+    return obtainInstruction("steak", 4);
+  })
+  .then((step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+    return obtainInstruction("steak", 5);
+  })
+  .then((step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+    return obtainInstruction("steak", 6);
+  })
+  .then((step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+
+    document.querySelector("#steak").innerHTML += `<li>Steak is ready!</li>`;
+
+    document.querySelector("#steakImg").removeAttribute("hidden");
+  })
+  .catch((error) => console.log(error));
+
+
 
 // Iteration 3 using async/await
-// ...
+async function makeBroccoli() {
+  // ... Your code here
+  const step1 = await obtainInstruction("broccoli", 0);
+  document.querySelector("#broccoli").innerHTML += `<li>${step1}</li>`;
```

Comment:
only thing I would change is avoid calling `document.querySelector` over and over. You could simply store into a variable and use that to change the innerHTML value.